addons/binding/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/internal/ssl/PinTrustManager.java

```diff
@@ -22,20 +21,19 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import javax.net.ssl.SSLEngine;
-import javax.net.ssl.X509ExtendedTrustManager;
+import javax.net.ssl.X509TrustManager;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 
 /**
- * This is a custom {@link X509ExtendedTrustManager}. {@link Pin} objects can be added and will
+ * This is a custom {@link X509TrustManager}. {@link Pin} objects can be added and will
  * be used in the checkServerTrusted() method to determine if a connection can be trusted.
  *
  * @author David Graeff - Initial contribution
  */
 @NonNullByDefault
-public class PinTrustManager extends X509ExtendedTrustManager {
+public class PinTrustManager implements X509TrustManager {
```

Comment:
Why did you downgrade to a `X509TrustManager`?

Reply:
Since interface of X509TrustManager was used only

Reply:
Yes that's reasonable. I haven't found the article I was looking for but for now look at this reference: https://log.rowanto.com/java-8-turning-off-ssl-certificate-check/

The JRE behaves differently depending on if you present a `X509ExtendedTrustManager` or a `X509TrustManager`. And we want the full control here and avoid framework guess work.